### PR TITLE
Fixes #3512. Ctrl+Z not suspending

### LINKED ISSUE #3512: `Ctrl+Z` not suspending
** Describe the bug**
`Ctrl+Z` not working in version `2.0.0-pre.`

**To Reproduce**
```cs
using Terminal.Gui;

static class Program
{
    class MainWindow : Toplevel
    {
        Label label;
        public MainWindow()
        {
            label = new Label
            {
                Text = "Hello, world!"
            };
            Add(label);
        }
    }

    static void Main(string[] args)
    {

        Application.Init();
        try
        {
            Application.Run<MainWindow>().Dispose();
        }
        finally
        {
            Application.Shutdown();
        }
    }
}

```
**Expected behavior**
When `Ctrl+Z` is pressed, the `app` should be suspended and returned to last state, when brought to foreground (`fg`).

**Desktop (please complete the following information):**
 - OS: macOS 14.4.1
 - Terminal: Terminal and iTerm2
 - Version: Terminal.Gui v2.0.0-pre.949

**Additional context**
This works fine in version `1.16.0`. I tried to test against the first `pre` versions of `2.0.0` and:
*** version <=`2.0.0-pre.231` ***
  - `Ctrl+Z` works always first time, but:
  - when returning to the `app`, `Ctrl+Z` and `Ctrl+Q` (`QuitKey`) don't work;
  - keyboard input is echoing
  - after exiting with `Ctrl+C`, mouse movements output *stuff* to the terminal.
 
*** version >= `2.0.0-pre.232` ***
  - `Ctrl+Z` does nothing.  
  - `OnKeyDown` is called.
 ```cs
        public override bool OnKeyDown(Key keyEvent)
        {
            label.Text = $"Keydown {keyEvent}";
            if (keyEvent == Key.Esc)
                Application.RequestStop();
            if (keyEvent == Key.Z.WithCtrl)
            {
                label.Text += " is Z with Ctrl";
                return false;
            }
            if (keyEvent == Key.Q.WithCtrl)
            {
                label.Text += " is Q with Ctrl";
                return false;
            }
            return true;
        }
```
if the code above is added to `MainWindow`, `Ctrl+Q` (`QuitKey`) is bubbled and Application exits but `Ctrl+Z` just updates `label.Text`.

### PR DESCRIPTION
**Pull request by BDisp**
## Fixes

- Fixes #3512

## Proposed Changes/Todos

- [x] Add #define UNIX_KEY_BINDINGS

## Pull Request checklist:

- [x] I've named my PR in the form of "Fixes #issue. Terse description."
- [x] My code follows the [style guidelines of Terminal.Gui](https://github.com/gui-cs/Terminal.Gui/blob/develop/.editorconfig) - if you use Visual Studio, hit `CTRL-K-D` to automatically reformat your files before committing.
- [x] My code follows the [Terminal.Gui library design guidelines](https://github.com/gui-cs/Terminal.Gui/blob/develop/CONTRIBUTING.md)
- [x] I ran `dotnet test` before commit
- [ ] I have made corresponding changes to the API documentation (using `///` style comments)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any poor grammar or misspellings
- [x] I conducted basic QA to assure all features are working

### CODE REVIEW

File: Terminal.Gui/Views/Toplevel.cs [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#define UNIX_KEY_BINDINGS
```
Comment (tig) [inline Terminal.Gui/Views/Toplevel.cs:1]: I do not want all those other "unix" key bindings enabled on non-Unix platforms. They get in the way of normal usage. They get in the way.

Until https://github.com/gui-cs/Terminal.Gui/issues/3023 is addressed (or something similarly flexible), can you please do this in this PR instead of `#define UNIX_KEY_BINDINGS`

```cs
if (Application.Driver is CursesDriver) 
{
   KeyBindings.Add (Key.Z.WithCtrl, Command.Suspend);
}
```

Thanks.

**Review by tig — CHANGES_REQUESTED** [review]

File: Terminal.Gui/Views/Toplevel.cs (hunk 1 above)
Comment (BDisp) [reply Terminal.Gui/Views/Toplevel.cs]: I changed to a conditional `PlatformID.Unix` because now `NetDriver` is also using the `Suspend` method on a `Linux` or `Mac`.

```cs
        if (Environment.OSVersion.Platform == PlatformID.Unix)
        {
            KeyBindings.Add (Key.Z.WithCtrl, Command.Suspend);
        }

```

**Review by dodexahedron — COMMENTED** [review]
Rather than hot loops, you can just start a free-running timer at startup and an event handler when it ticks.

Should be like... half the code, if even that much.

File: Terminal.Gui/ConsoleDrivers/ConsoleDriver.cs [hunk 2]
```diff
@@ -315,12 +315,31 @@ public void ClearContents ()
...
+                _dirtyLines [row] = true;
+            }
+        }
+    }
+
+    /// <summary>
+    /// Sets <see cref="Contents"/> as dirty for situations where views
+    /// don't need layout and redrawing, but just refresh the screen.
+    /// </summary>
+    public void SetContentsAsDirty ()
```
Comment (tig) [inline Terminal.Gui/ConsoleDrivers/ConsoleDriver.cs:332]: 1) What does this have to do with Ctrl-Z? 
2) I think this would be better implemented as 

```cs
public bool ForceFullUpdate {get;set;}
```

Then in `Update`:

```cs
      if (ForceFullUpdate || !Contents [row, col].IsDirty)
...

ForceFullUpdate = false;
```

Comment (tig) [reply Terminal.Gui/ConsoleDrivers/ConsoleDriver.cs:332]: And can't it be made `private` since it's only called from a derived class?

**Review by tig — CHANGES_REQUESTED** [review]
Good stuff. See my comments.

File: Terminal.Gui/ConsoleDrivers/ConsoleDriver.cs (hunk 2 above)
Comment (BDisp) [reply Terminal.Gui/ConsoleDrivers/ConsoleDriver.cs:332]: > 1. What does this have to do with Ctrl-Z?

This is related with the issue #3525 and was intended to use in the `Suspend` method, so it have to do with Ctrl-Z and I leveraged this branch to implement.

> 2. I think this would be better implemented as
> 
> ```cs
> public bool ForceFullUpdate {get;set;}
> ```
> 
> Then in `Update`:
> 
> ```cs
>       if (ForceFullUpdate || !Contents [row, col].IsDirty)
> ...
> 
> ForceFullUpdate = false;
> ```

I don't agree because if it use a property I have to include that code on two derived class, `CursesDriver` and `NetDriver`. Using a method on the `ConsoleDriver` it's only implemented once. If you don't like the method name, then you may suggest another name.

Comment (BDisp) [reply Terminal.Gui/ConsoleDrivers/ConsoleDriver.cs:332]: > And can't it be made `private` since it's only called from a derived class?

No, because it is inaccessible in derived classes if I change to private.

File: Terminal.Gui/Views/Toplevel.cs (hunk 1 above)
Comment (tig) [reply Terminal.Gui/Views/Toplevel.cs]: Thanks!

File: Terminal.Gui/ConsoleDrivers/ConsoleDriver.cs (hunk 2 above)
Comment (tig) [reply Terminal.Gui/ConsoleDrivers/ConsoleDriver.cs:332]: Sounds good. I appreciate the explanation; makes sense.

**Review by tig — APPROVED** [review]
